src/main/java/io/aiven/kafka/auth/AivenAclAuthorizer.java

```diff
@@ -44,33 +53,17 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(AivenAclAuthorizer.class);
     private File configFile;
-    private ReentrantReadWriteLock lock = new ReentrantReadWriteLock(true);
-    private long lastUpdateCheckTimestamp = 0;
-    private long lastModifiedTimestamp = 0;
-    private List<AivenAcl> aclEntries;
-    private Map<String, Boolean> verdictCache;
+    private final AtomicReference<List<AivenAcl>> aclEntries = new AtomicReference<>();
+    private final Map<String, Boolean> verdictCache = new ConcurrentHashMap<>();
     private JsonReader<AivenAcl> jsonReader;
     private AuditorAPI auditor;
     private boolean logDenials;
+    private final ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+    private WatchService watchService;
```

Comment:
Can be initialized here (`= FileSystems.getDefault().newWatchService();`) and be `final`.

Reply:
It can't because of checked exception.